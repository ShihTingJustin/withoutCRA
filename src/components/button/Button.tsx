import React from 'react';
import useLongPress from '../../hooks/useLongPress';

import './button.scss';

const Button = ({
  children,
  disabled,
  onClick
}: {
  children: React.ReactNode;
  disabled?: boolean;
  onClick: () => void;
}) => {
  const { handleMouseDown, handleMouseUp } = useLongPress({ onClick });

  return (
    <div data-testid="button" className="button-root">
      <button
        data-disabled={disabled}
        onClick={onClick}
        onMouseDown={handleMouseDown}
        onMouseUp={handleMouseUp}
        onMouseLeave={handleMouseUp}
      >
        {children}
      </button>
    </div>
  );
};

export default Button;
