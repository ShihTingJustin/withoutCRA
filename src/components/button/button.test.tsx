import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import Button from './Button';
import { AiOutlineMinus } from 'react-icons/ai';

describe('Test Button component', () => {
  const component = (
    <Button onClick={() => {}}>
      <span>
        <AiOutlineMinus size={30} color="#1e9fd2" />
      </span>
    </Button>
  );

  it('[Render] match snapshot', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('[Render] toBeInTheDocument', () => {
    render(component);
    const element = screen.getByTestId('button');
    expect(element).toBeInTheDocument();
  });
});
