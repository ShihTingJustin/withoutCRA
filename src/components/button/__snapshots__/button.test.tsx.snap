// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component [Render] match snapshot 1`] = `
<div
  className="button-root"
  data-testid="button"
>
  <button
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
  >
    <span>
      <svg
        color="#1e9fd2"
        fill="currentColor"
        height={30}
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": "#1e9fd2",
          }
        }
        viewBox="0 0 1024 1024"
        width={30}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M872 474H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h720c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
        />
      </svg>
    </span>
  </button>
</div>
`;
